File: totvs-camera-core/src/main/java/com/totvs/camera/core/ImageInfo.kt
package com.totvs.camera.core

/** Metadata for an image. */
interface ImageInfo {
    /**
     * Timestamp of the taken image
     */
    val timestamp: Long

    /**
     * Returns the rotation needed to transform the image to the correct orientation.
     *
     * This is a clockwise rotation in degrees that needs to be applied to the image buffer.
     * Note that for images that are in [android.graphics.ImageFormat.JPEG] this value will
     * match the rotation defined in the EXIF.
     *
     * The target rotation is set at the time the image capture was requested.
     */
    val rotationDegrees: Int
}

File: ios/TOTVSCameraKit/Core/ImageInfo.swift
//
//  ImageInfo.swift
//  TOTVSCameraKit
//
//  Created by Jansel Rodriguez on 7/8/20.
//  Copyright © 2020 Jansel. All rights reserved.
//

import Foundation
import UIKit

/** Metadata for an image. */
public protocol ImageInfo {
    
    /**
     @property timestamp
     @abstract
        Timestamp of the taken image.
    */
    var timestamp: Int64 { get }
    
    /**
     @property orientation
     @abstract
        Returns the orientation needed to transform the image to the correct orientation.
     
     @discussion
        The target orientation is set at the time the image capture was requested.
     */
    var orientation: UIDeviceOrientation { get }

    /**
     @property sourceFacing
     @abstract
        Returns the source camera facing that was used to capture this image
     */
    var sourceFacing: CameraFacing { get }
}
